### ISSUE TITLE
Common Exogenous Regressors like in R Fable package

### ISSUE THREAD

**Issue by brianfhead** [Nixtla/statsforecast#633]
### Description

I'd love to see the funcationality in StatsForecast and MLforecast to use what is termed "common xregs" in Fable (see the link below)--trend, season, and fourier. 

https://fable.tidyverts.org/reference/common_xregs.html

Thanks!

**Comment by brianfhead** [comment]
Here's some additional info that could be used for a "use case"...copied from a post/thread I put on the Nixtla Slack #general channel: 

Recently I included in a thread/post that I was getting substantially better results with R fable including prophet model than I could with SF and MF. With SF/MF/NF I followed the deseasonalizing approach whereas with fable I used the built in "common exogenous" variables trend and season. I decided to test using trend/season in the Nixtla approach so I extracted those using the statsmodels package. The results of that test suggest it wasn't the prophet model driving the difference, but that, at least with my current data, using the trend/season worked better than deseasonalizing. For example, with approximately 1200 series I compared the MASE for the best model from R fable and SF/MF/NF. Went from fable having the lower MASE ~65% of the time to SF/MF/NF having it ~70% of the time. Similarly, I went from having a MASE below 0.5 (twice as good as a Naive model) for SF/MF/NF for ~45% of series to 82%.
Interestingly, even though I included AutoARIMA with fable, prophet "won" with the lowest MASE the vast majority of the time when working in R. BUT, with the Nixtla packages the AutoARIMA package won a solid majority of the time, beating out all of the available SF models, most of the regressors available in MF, and four neural models--supporting other work from Nixtla that traditional and MF models frequently best deep learning.
Really appreciate all of the feedback I've received from this community/the Nixtla team to help me transition from fable to the Nixtlaverse.
